# Q-1

1-Suppose that we would like to know how much families in the US are spending on recreation annually. We estimated the following model:

expenditure=873+0.0012annual_income+0.00002annual_income2−223.57have_kids
 
expenditure is the annual spending on recreation in US dollars, annual_income is the annual income in US dollars and have_kids is a dummy variable indicating the families with children. Interpret the estimated coefficients. What additional statistics should be given in order to make sure that your interpretations make sense statistically. Write up your answer and save in a Gist.

- 873 sabiti tüm değişkenler 0 değerini aldığında harcamanın alacağı değeri verir. Sabiti koymamızın amacı, dengelemek.

- Yıllık gelir pozitif yönlü olduğu görülmektedir. Yıllık gelir 1 arttığında 0.0012si kadar eğlence harcamasının artacağını gösterir.

- Yıllık harcamanın karesi eğlence harcamalar üzerinde daha küçük bir artış olacağını gösteriyor. 

- Çocuk sayısındaki 1 artışın ise eğelnce harcamalarını 223 azaltacağı yönünde.

- Bu katsayılara bakarak hangisinin en çok etkili olduğu konusunda bir yargıya varamayız. 

# Q-2

2-First, load the dataset from the weatherinszeged table from Kaggle.

Build a linear regression model where your target variable is the difference between the apparenttemperature and the temperature. As explanatory variables, use humidity and windspeed. Now, estimate your model using OLS. Are the estimated coefficients statistically significant? Are the signs of the estimated coefficients in line with your previous expectations? Interpret the estimated coefficients. What are the relations between the target and the explanatory variables?

Next, include the interaction of humidity and windspeed to the model above and estimate the model using OLS. Are the coefficients statistically significant? Did the signs of the estimated coefficients for humidity and windspeed change? Interpret the estimated coefficients.


In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from  scipy.stats import zscore
import seaborn as sns
from sklearn import linear_model
import statsmodels.api as sm
import warnings
warnings.filterwarnings('ignore')
pd.options.display.max_columns = None

In [6]:
weather = pd.read_csv("weatherHistory.csv")
weather.head()

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   

                       Daily Summary  
0  Partly cloudy throughout the day.  
1  Partly cloudy throughout the day.  
2  Partly cloudy throughout the day.  
3  Partly cloudy throughout the day.  
4  Partly cloudy throughout the day.

In [7]:
print(weather.isnull().sum())
print("**"*22)
print(weather.isnull().sum()*100/weather.shape[0])


Formatted Date                0
Summary                       0
Precip Type                 517
Temperature (C)               0
Apparent Temperature (C)      0
Humidity                      0
Wind Speed (km/h)             0
Wind Bearing (degrees)        0
Visibility (km)               0
Loud Cover                    0
Pressure (millibars)          0
Daily Summary                 0
dtype: int64
********************************************
Formatted Date              0.000000
Summary                     0.000000
Precip Type                 0.536012
Temperature (C)             0.000000
Apparent Temperature (C)    0.000000
Humidity                    0.000000
Wind Speed (km/h)           0.000000
Wind Bearing (degrees)      0.000000
Visibility (km)             0.000000
Loud Cover                  0.000000
Pressure (millibars)        0.000000
Daily Summary               0.000000
dtype: float64


In [8]:
weather.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96453 entries, 0 to 96452
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Formatted Date            96453 non-null  object 
 1   Summary                   96453 non-null  object 
 2   Precip Type               95936 non-null  object 
 3   Temperature (C)           96453 non-null  float64
 4   Apparent Temperature (C)  96453 non-null  float64
 5   Humidity                  96453 non-null  float64
 6   Wind Speed (km/h)         96453 non-null  float64
 7   Wind Bearing (degrees)    96453 non-null  float64
 8   Visibility (km)           96453 non-null  float64
 9   Loud Cover                96453 non-null  float64
 10  Pressure (millibars)      96453 non-null  float64
 11  Daily Summary             96453 non-null  object 
dtypes: float64(8), object(4)
memory usage: 8.8+ MB


In [9]:
#Target Variable
weather["TempatureDif"] = weather["Temperature (C)"] - weather["Apparent Temperature (C)"]

Y = weather["TempatureDif"]

X = weather[["Humidity","Wind Speed (km/h)"]]
X = sm.add_constant(X)

results = sm.OLS(Y,X).fit()
display(results.summary())

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           TempatureDif   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Tue, 15 Sep 2020   Prob (F-statistic):               0.00
Time:                        15:39:17   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -2.4381      0.021   -115.948      0.000      -2.479      -2.397
Humidity              3.0292      0.024    126.479      0.000       2.982       3.076
Wind Speed (km/h)     0.1193      0.001    176.164      0.000       0.118       0.121
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                           0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Sonuçlara göre bulunan değişkenlerin katsayıları 0.05 anlam düzeyinden küçük oldukları için istatistiksel olarak anlamlı olduğunu söyleyebiliriz.

Katsayılara bakacak olursak modeli şu şekilde oluşturabiliriz. 

TempatureDif = -2.4381 + 3.0292Humidity + 0.1193Wind Speed (km/h)

Dengeleyici sabit -2.4381 değerini almıştır. 

Nem bir birim arttığında tempatureDif 3.029 artmaktadır. 

Rüzgar hızı ise bir birim arttığında TempatureDif 0.1193 artmaktadır. 

Bu katsayılara bakarak hangisinin en çok etkili olduğu anlaşılmamaktadır. 

# Q-3

3-Load the houseprices data from Kaggle.

Run your house prices model again and interpret the results. Which features are statistically significant and which are not?

Now, exclude the insignificant features from your model. Did anything change?

Interpret the statistically significant coefficients by quantifying their relations with the house prices. Which features have more prominent effect on the house prices?

Do the results sound reasonable to you? If not, try to explain the potential reasons.


In [10]:
house_train = pd.read_csv("house_train.csv")
house_train.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities LotConfig LandSlope Neighborhood Condition1  \
0         Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
1         Lvl    AllPub       FR2       Gtl      Veenker      Feedr   
2         Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
3         Lvl    AllPub    Corner       Gtl      Crawfor       Norm   
4         Lvl    AllPub       FR2       Gtl      NoRidge       Norm   

  Condition2 BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  \
0       Norm     1Fam     2Story            7            5       2003   
1       Norm     1Fam     1Story            6            8       1976   
2       Norm     1Fam     2Story            7            5       2001   
3       Norm     1Fam     2Story            7            5       1915   
4       Norm     1Fam     2Story            8            5       2000   

   YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd MasVnrType  \
0          2003     Gable  CompShg     VinylSd     VinylSd    BrkFace   
1          1976     Gable  CompShg     MetalSd     MetalSd       None   
2          2002     Gable  CompShg     VinylSd     VinylSd    BrkFace   
3          1970     Gable  CompShg     Wd Sdng     Wd Shng       None   
4          2000     Gable  CompShg     VinylSd     VinylSd    BrkFace   

   MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond BsmtExposure  \
0       196.0        Gd        TA      PConc       Gd       TA           No   
1         0.0        TA        TA     CBlock       Gd       TA           Gd   
2       162.0        Gd        TA      PConc       Gd       TA           Mn   
3         0.0        TA        TA     BrkTil       TA       Gd           No   
4       350.0        Gd        TA      PConc       Gd       TA           Av   

  BsmtFinType1  BsmtFinSF1 BsmtFinType2  BsmtFinSF2  BsmtUnfSF  TotalBsmtSF  \
0          GLQ         706          Unf           0        150          856   
1          ALQ         978          Unf           0        284         1262   
2          GLQ         486          Unf           0        434          920   
3          ALQ         216          Unf           0        540          756   
4          GLQ         655          Unf           0        490         1145   

  Heating HeatingQC CentralAir Electrical  1stFlrSF  2ndFlrSF  LowQualFinSF  \
0    GasA        Ex          Y      SBrkr       856       854             0   
1    GasA        Ex          Y      SBrkr      1262         0             0   
2    GasA        Ex          Y      SBrkr       920       866             0   
3    GasA        Gd          Y      SBrkr       961       756             0   
4    GasA        Ex          Y      SBrkr      1145      1053             0   

   GrLivArea  BsmtFullBath  BsmtHalfBath  FullBath  HalfBath  BedroomAbvGr  \
0       1710             1             0         2         1             3   
1       1262             0             1         2         0             3   
2       1786             1             0         2         1             3   
3       1717             1             0         1         0             3   
4       2198             1             0         2         1             4   

   KitchenAbvGr KitchenQual  TotRmsAbvGrd Functional  Fireplaces FireplaceQu  \
0             1          Gd             8        Typ           0         NaN   
1             1          TA             6        Typ           1          TA   
2             1          Gd             6        Typ           1          TA   
3             1          Gd            

In [21]:
house_train.columns

Index(['Id', 'MSSubClass', 'MSZoning', 'LotArea', 'Street', 'LotShape',
       'LandContour', 'Utilities', 'LotConfig', 'LandSlope', 'Neighborhood',
       'Condition1', 'Condition2', 'BldgType', 'HouseStyle', 'OverallQual',
       'OverallCond', 'YearBuilt', 'YearRemodAdd', 'RoofStyle', 'RoofMatl',
       'Exterior1st', 'Exterior2nd', 'MasVnrType', 'MasVnrArea', 'ExterQual',
       'ExterCond', 'Foundation', 'BsmtQual', 'BsmtCond', 'BsmtExposure',
       'BsmtFinType1', 'BsmtFinSF1', 'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF',
       'TotalBsmtSF', 'Heating', 'HeatingQC', 'CentralAir', 'Electrical',
       '1stFlrSF', '2ndFlrSF', 'LowQualFinSF', 'GrLivArea', 'BsmtFullBath',
       'BsmtHalfBath', 'FullBath', 'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr',
       'KitchenQual', 'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual',
       'GarageCond', 'PavedDrive', 'WoodDeckSF', 'OpenPorchSF',
       'Enclos

In [11]:
pd.options.display.max_rows
house_train.isnull().sum()
print(house_train.isnull().sum()*100/house_train.shape[0])

Id                0.000000
MSSubClass        0.000000
MSZoning          0.000000
LotFrontage      17.739726
LotArea           0.000000
                   ...    
MoSold            0.000000
YrSold            0.000000
SaleType          0.000000
SaleCondition     0.000000
SalePrice         0.000000
Length: 81, dtype: float64


In [12]:
house_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

In [13]:
house_train.drop(['PoolQC', "Fence", "MiscFeature","FireplaceQu","Alley","LotFrontage"],axis=1,inplace=True)

In [14]:
print(house_train.isnull().sum()*100/house_train.shape[0])

Id               0.0
MSSubClass       0.0
MSZoning         0.0
LotArea          0.0
Street           0.0
                ... 
MoSold           0.0
YrSold           0.0
SaleType         0.0
SaleCondition    0.0
SalePrice        0.0
Length: 75, dtype: float64


In [18]:
house_train["GarageYrBlt"].fillna(house_train["GarageYrBlt"].mode()[0],inplace=True)
house_train["GarageCond"].fillna(house_train["GarageCond"].mode()[0],inplace=True)
house_train["GarageFinish"].fillna(house_train["GarageFinish"].mode()[0],inplace=True)
house_train["GarageArea"].fillna(house_train["GarageArea"].mode()[0],inplace=True)
house_train["GarageQual"].fillna(house_train["GarageQual"].mode()[0],inplace=True)
house_train["GarageCars"].fillna(house_train["GarageCars"].mode()[0],inplace=True)
house_train["GarageType"].fillna(house_train["GarageType"].mode()[0],inplace=True)

house_train.dropna(inplace=True)

Öncelikle hangileri ile ilgili olabilir diye yaptığım incelemeden sonra --> Steet,GarageQual,GarageCars,CentralAir,HeatingQC, TotRmsAbvGrd,GarageArea

In [31]:
house_train["central_air"]=pd.get_dummies(house_train["CentralAir"], drop_first=True)

house_train["Street_"]=pd.get_dummies(house_train["Street"], drop_first=True)

In [22]:
def heatingquc(HeatingQC):
    if HeatingQC=="Ex":
        return 5
    elif HeatingQC=="Gd":
        return 4
    elif HeatingQC=="TA":
        return 3
    elif HeatingQC=="Fa":
        return 2
    elif HeatingQC=="Po":
        return 1


house_train["HeatingQc"] = house_train["HeatingQC"].apply(heatingquc)    

In [24]:
def Garagequal(GarageQual):
    if GarageQual=="Ex":
        return 6
    elif GarageQual=="Gd":
        return 5
    elif GarageQual=="TA":
        return 4
    elif GarageQual=="Fa":
        return 3
    elif GarageQual=="Po":
        return 2
    elif GarageQual=="NA":
        return 1


house_train["GarageQuall"] = house_train["GarageQual"].apply(Garagequal)    

In [25]:
def kitchenqual(KitchenQual):
    if KitchenQual=="Ex":
        return 4
    elif KitchenQual=="Gd":
        return 3
    elif KitchenQual=="TA":
        return 2
    elif KitchenQual=="Fa":
        return 1

house_train["Kitchen_Qual"]=house_train["KitchenQual"].apply(kitchenqual)

In [32]:
Y_ = house_train["SalePrice"]

X_ = house_train[["HeatingQc","GarageQuall","GarageCars",
                 "central_air","TotRmsAbvGrd","GarageArea","Kitchen_Qual","GrLivArea","Street_"]]
X_ = sm.add_constant(X_)

results = sm.OLS(Y_,X_).fit()
display(results.summary())

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.717
Model:                            OLS   Adj. R-squared:                  0.715
Method:                 Least Squares   F-statistic:                     394.6
Date:                Tue, 15 Sep 2020   Prob (F-statistic):               0.00
Time:                        15:47:49   Log-Likelihood:                -17040.
No. Observations:                1412   AIC:                         3.410e+04
Df Residuals:                    1402   BIC:                         3.415e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const         -1.49e+05   2.57e+04     -5.791      0.000   -1.99e+05   -9.85e+04
HeatingQc     5113.7035   1396.274      3.662      0.000    2374.692    7852.715
GarageQuall   1.452e+04   4704.238      3.087      0.002    5295.348    2.38e+04
GarageCars    2.066e+04   3300.686      6.259      0.000    1.42e+04    2.71e+04
central_air   1.707e+04   5192.120      3.287      0.001    6882.173    2.73e+04
TotRmsAbvGrd -2256.2384   1256.372     -1.796      0.073   -4720.809     208.332
GarageArea      24.3935     11.456      2.129      0.033       1.921      46.866
Kitchen_Qual   3.42e+04   2226.830     15.360      0.000    2.98e+04    3.86e+04
GrLivArea       73.0757      4.170     17.526      0.000      64.897      81.255
Street_       5573.7117   1.75e+04      0.318      0.750   -2.88e+04    3.99e+04
==============================================================================
Omnibus:                      372.810   Durbin-Watson:                   2.038
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            11489.848
Skew:                           0.564   Prob(JB):                         0.00
Kurtosis:                      16.929   Cond. No.                     4.32e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.32e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Sonuçlara göre;

GarageQuall: Garaj kalitesindeki bir birimlik artış 1.731e+04 fiyat üzerinde artma görülmektedir.

GarageCars: Garajın aldığı araba sayısındaki bir birimlik artış 2.134e+04 fiyat üzerinde artma görülmektedir.

central_air: Merkezi klimanın olması 1.979e+04 kadar etkilemektedir.

TotRmsAbvGrd: Oda sayısındaki bir birimlik artış fiyatı 1.537e+04 etkilemektedir.

GarageArea: Garaj alanındaki bir birimlik artış fiyatı 49.4966 arttırmaktadır.

Kitchen_Qual: Kalitedeki bir birimlik iyileşme fiyatı 4.458e+04	arttırmaktadır. 


P değerleri incelendiği zaman 0.05 anlam düzeyinden büyük olanların hedef değişkeni etkilemediği görülmektedir. Bu yüzden çıkarılmalıdır. 